azurerm/resource_arm_firewall.go

```diff
@@ -62,24 +62,22 @@ func resourceArmFirewall() *schema.Resource {
 						},
 						"subnet_id": {
 							Type:         schema.TypeString,
-							Required:     true,
+							Optional:     true,
```

Comment:
Is there a reason for changing this to `Optional`? I think this should still be `Required`, WDYT?

Reply:
Yes, the reason is RestApi requires only one `subnet_id` in `ip_configuration`. 0 or more than 1 `subnet_id` will cause errors.

Now, my solution is to change the `subnet_id` to `Optional` and check it in our new introduced `validateFirewallConfigurationSettings` function to make sure there is only one `subnet_id ` in `ip_configuration`.

I have considered other ways like move `subnet_id` outside from `ip_configuration`, but it may make users confused and also bring breaking changes.

What do you think?

Reply:
the documentation needs to be updated to match this being Optional too: https://github.com/terraform-providers/terraform-provider-azurerm/pull/4639/files#diff-0f2b98bda27ee42e45f9395c8f830b0dR77